packages/amplify-category-function/src/provider-utils/awscloudformation/utils/functionPluginLoader.ts

```diff
@@ -47,46 +52,87 @@ export async function templateWalkthrough(context: any, params: Partial<Function
 }
 
 /**
- * Selects a function runtime
+ * Selects a single runtime for a Lambda Function
+ * Selects one or more runtimes for a Lambda Layer
  */
 export async function runtimeWalkthrough(
   context: any,
-  params: Partial<FunctionParameters>,
-): Promise<Pick<FunctionParameters, 'runtimePluginId'> & FunctionRuntimeParameters> {
+  params: Partial<FunctionParameters> | Partial<LayerParameters>,
+): Promise<Array<Pick<FunctionParameters, 'runtimePluginId'> & FunctionRuntimeParameters>> {
+  const { service } = params.providerContext;
   const selectionOptions: PluginSelectionOptions<FunctionRuntimeCondition> = {
     pluginType: 'functionRuntime',
     listOptionsField: 'runtimes',
     predicate: condition => {
-      return condition.provider === params.providerContext.provider && condition.service === params.providerContext.service;
+      return condition.provider === params.providerContext.provider && condition.services.includes(service);
     },
-    selectionPrompt: 'Choose the function runtime that you want to use:',
+    selectionPrompt:
+      service === ServiceName.LambdaLayer ? 'Select up to 5 compatible runtimes:' : 'Choose the runtime that you want to use:',
     notFoundMessage: `No runtimes found for provider ${params.providerContext.provider} and service ${params.providerContext.service}`,
+    service,
   };
-  const selection = await getSelectionFromContributors<FunctionRuntimeCondition>(context, selectionOptions);
-  const plugin = await loadPluginFromFactory(selection.pluginPath, 'functionRuntimeContributorFactory', context);
-  const depCheck = await (plugin as FunctionRuntimeLifecycleManager).checkDependencies(selection.value);
-  if (!depCheck.hasRequiredDependencies) {
-    context.print.warning(depCheck.errorMessage || 'Some dependencies required for building and packaging this runtime are not installed');
+  const selections = await getSelectionFromContributors<FunctionRuntimeCondition>(context, selectionOptions);
+  const plugins = [];
+  for (let selection of selections) {
+    const plugin = await loadPluginFromFactory(selection.pluginPath, 'functionRuntimeContributorFactory', context);
+    const depCheck = await (plugin as FunctionRuntimeLifecycleManager).checkDependencies(selection.value);
+    if (!depCheck.hasRequiredDependencies) {
+      context.print.warning(
+        depCheck.errorMessage || 'Some dependencies required for building and packaging this runtime are not installed',
+      );
+    }
+    plugins.push(plugin);
+  }
+  if (service === ServiceName.LambdaFunction) {
+    return _functionRuntimeWalkthroughHelper(params, plugins[0], selections[0]);
+  } else if (service === ServiceName.LambdaLayer) {
+    return _layerRuntimeWalkthroughHelper(params, plugins, selections);
   }
-  const contributionRequest: ContributionRequest = {
+}
+
+async function _functionRuntimeWalkthroughHelper(
```

Comment:
couldn't this function and the layer helper below be combined? If you pass a plugin and selection array with one element to the layer helper, I think you get the same thing as if you had used the function helper

Reply:
I have includes this change , It will come up in my PR